modules/cxf-client-rest/src/main/java/com/devonfw/module/cxf/common/impl/client/rest/RestServiceExceptionMapper.java

```diff
@@ -45,13 +53,83 @@ public Throwable fromResponse(Response response) {
       String data = response.readEntity(String.class);
       if ((data != null) && !data.isEmpty()) {
         MediaType mediaType = response.getMediaType();
-        URI url = response.getLocation();
-        String operation = null;
-        String serviceDetails = this.context.getServiceDescription(operation, url.toString());
+        String url = getUrl(response);
+        String operation = getOperation(response);
+        String serviceDetails = this.context.getServiceDescription(operation, url);
         return this.errorUnmarshaller.unmarshall(data, mediaType.toString(), response.getStatus(), serviceDetails);
       }
     }
     return null;
   }
 
+  private String getOperation(Response response) {
+
+    if (response instanceof ResponseImpl) {
+      Message message = ((ResponseImpl) response).getOutMessage();
+      if (message != null) {
+        Object invocationContext = message.get(Message.INVOCATION_CONTEXT);
+        Object requestContext = getFromMap(invocationContext, "RequestContext");
+        OperationResourceInfo operation = getFromMap(requestContext, OperationResourceInfo.class);
+        if (operation != null) {
+          Method method = operation.getAnnotatedMethod();
+          if (method != null) {
+            return method.getName();
+          }
+        }
+      }
+    }
+    return null;
+  }
+
+  @SuppressWarnings("rawtypes")
+  private static Object getFromMap(Object map, Object key) {
+
+    if (map instanceof Map) {
+      return ((Map) map).get(key);
+    }
+    return null;
+  }
+
+  @SuppressWarnings("rawtypes")
+  private static <T> T getFromMap(Object map, Class<T> key) {
+
+    if (map instanceof Map) {
+      Object value = ((Map) map).get(key.getName());
+      if (value != null) {
+        try {
+          return key.cast(value);
+        } catch (Exception e) {
+        }
+      }
+    }
+    return null;
+  }
+
+  private String getUrl(Response response) {
+
+    URI url = response.getLocation();
+    if (url != null) {
+      return url.toString();
+    } else if (response instanceof ResponseImpl) {
+      Message message = ((ResponseImpl) response).getOutMessage();
+      if (message != null) {
+        Object uri = message.get(Message.REQUEST_URI);
+        if (uri instanceof String) {
+          return (String) uri;
+        }
+        Conduit conduit = message.get(Conduit.class);
+        if (conduit != null) {
+          EndpointReferenceType target = conduit.getTarget();
+          if (target != null) {
+            AttributedURIType address = target.getAddress();
+            if (address != null) {
+              return address.getValue();
+            }
+          }
+        }
+      }
+    }
+    return "url-not-available-in-cxf-response";
```

Comment:
maybe we should return null instead?

Reply:
hopefully we should never get here either...

Reply:
maybe return "" is even safer to avoid NPE.

Reply:
Its a private method that is only used in a single place to pass it to `getServiceDescription` here:
https://github.com/devonfw/devon4j/blob/0ac5c96673b4c5c2b08ab73b93692aac41b52ad3/modules/cxf-client-rest/src/main/java/com/devonfw/module/cxf/common/impl/client/rest/RestServiceExceptionMapper.java#L58
That method honors `null` but not the empty String as the idea was that `null` means not available:
https://github.com/devonfw/devon4j/blob/0ac5c96673b4c5c2b08ab73b93692aac41b52ad3/modules/service/src/main/java/com/devonfw/module/service/common/api/client/context/ServiceContext.java#L63

However, we could also return "" and also change `getServiceDescription` to treat "" just like `null`.